components/tidb_query_expr/src/types/expr_builder.rs

```diff
@@ -395,6 +398,16 @@ fn extract_scalar_value_int64(val: Vec<u8>) -> Result<ScalarValue> {
     Ok(ScalarValue::Int(Some(value)))
 }
 
+#[inline]
+fn extract_scalar_value_uint64_from_bits(val: Vec<u8>) -> Result<ScalarValue> {
+    debug_assert!(val.len() <= 8);
+    let mut res = 0;
+    for (i, v) in val.iter().enumerate() {
+        res |= (*v as u64) << ((val.len() - i - 1) * 8);
+    }
```

Comment:
If I don't think wrong, could the logic to be:
```rust
let mut res = 0;
for v in val {
    res <<= 8;
    res |= v as u64;
}
```

Reply:
right, yours is more clear, updated.